src/main/java/com/fcgl/Listing/Vendors/Amazon/Feed/GenerateFeedSubmissionList/GenerateFeedSubmissionList.java

```diff
@@ -0,0 +1,197 @@
+            builder.append(feedSubmissionInfo.getFeedSubmissionId());
+            builder.append("\n");
+          }
+
+          if (feedSubmissionInfo.isSetFeedType()) {
+            builder.append("                FeedType\n                    ");
+            builder.append(feedSubmissionInfo.getFeedType());
+            builder.append("\n");
+            
+          }
+
+          if (feedSubmissionInfo.isSetSubmittedDate()) {
+            builder.append("                SubmittedDate\n                    ");
+            builder.append(feedSubmissionInfo.getSubmittedDate());
+            builder.append("\n");
+          }
+
+          if (feedSubmissionInfo.isSetFeedProcessingStatus()) {
+            String feedProcessingStatus = feedSubmissionInfo.getFeedProcessingStatus();
+            builder.append("                FeedProcessingStatus\n                    ");
+            builder.append(feedProcessingStatus);
+            builder.append("\n");
+            feedStatus.add(feedProcessingStatus);
+          }
+
+          if (feedSubmissionInfo.isSetStartedProcessingDate()) {
+            builder.append("                StartedProcessingDate\n                    ");
+            builder.append(feedSubmissionInfo.getStartedProcessingDate());
+            builder.append("\n");
+          }
+
+          if (feedSubmissionInfo.isSetCompletedProcessingDate()) {
+            builder.append("                CompletedProcessingDate\n                    ");
+            builder.append(feedSubmissionInfo.getCompletedProcessingDate());
+            builder.append("\n");
+          }
+        }
+      }
+
+      String amzRequestId = "";
+      if (response.isSetResponseMetadata()) {
+        builder.append("        ResponseMetadata\n");
+        ResponseMetadata responseMetadata = response.getResponseMetadata();
+        if (responseMetadata.isSetRequestId()) {
+          amzRequestId = responseMetadata.getRequestId();
+          builder.append("            RequestId\n                ");
+          builder.append(amzRequestId);
+          builder.append("\n");
+        }
+      }
+
+      builder.append("\n");
+      builder.append(response.getResponseHeaderMetadata());
+      builder.append("\n\n");
+      logger.info(builder.toString());
+      return feedStatus;
+  }
+
+  /**
+   * Adds a message to a queue based on the feed processing status that Amazon returns.
+   * _DONE_:          Should be sent to the PROCESSING_COMPLETE queue to continue the Amazon workflow
+   * _CANCELLED_:     Should be sent to the FEED_CANCELED queue to investigate why there was a fatal error.
+   * _IN_SAFETY_NET_: Should be sent to a SAFETY_NET queue to investigate
+   * Default:         Should be requeued because the feedSubmission is still processing
+   * @param statuses: Statuses returned by Amazon
+   */
+  private void handleFeedProcessingStatus(List<String> statuses, List<String> submissions) {
```

Comment:
It'd be nice to get this under some unit tests. Perhaps add an interface and make this public in it.